flexbox/src/main/java/com/google/android/flexbox/FlexboxLayoutManager.java

```diff
@@ -368,8 +368,10 @@ public void setAlignContent(@AlignContent int alignContent) {
 
     @Override
     public int getDecorationLengthMainAxis(View view, int index, int indexInFlexLine) {
+        View v = mRecycler.getViewForPosition(index);
+        calculateItemDecorationsForChild(v, TEMP_RECT);
         if (isMainAxisDirectionHorizontal()) {
-            return getLeftDecorationWidth(view) + getRightDecorationWidth(view);
+            return getLeftDecorationWidth(v) + getRightDecorationWidth(v);
```

Comment:
I really don't like this solution. Why? You're creating new `View` instance instead of reusing the existing one.  

Code from lines _371_ and _372_ you should put into _if_ conditional statement, but instead of using new `v` variable, please use exisitng `view`.

To be more precise:

```
if (isMainAxisDirectionHorizontal()) {
         view = mRecycler.getViewForPosition(index);
         calculateItemDecorationsForChild(v, TEMP_RECT);
         return getLeftDecorationWidth(v) + getRightDecorationWidth(v);
} else {
          return getTopDecorationHeight(view) + getBottomDecorationHeight(view);
}
```